delta/migration/src/main/scala/ch/epfl/bluebrain/nexus/migration/Migration.scala

```diff
@@ -0,0 +1,514 @@
+              fixResolverSource(source).flatMap { s =>
+                updateResolver(s)
+                  .onErrorRecover { case _: ResolverRejection.InvalidJsonLdFormat =>
+                    updateResolver(fixId(s, id))
+                  }
+                  .toTask
+                  .void
+              }
+            case Deprecated(id, _, _, rev, types, _, _) if types.contains(nxv.Resolver)    =>
+              resolvers.deprecate(IriSegment(id), projectRef, rev - 1).toTask.void
+            //TODO Views
+            case Created(_, _, _, _, types, _, _, _) if types.contains(ViewType)           =>
+              IO.unit
+            case Updated(_, _, _, _, types, _, _, _) if types.contains(ViewType)           =>
+              IO.unit
+            case Deprecated(_, _, _, _, types, _, _) if types.contains(ViewType)           =>
+              IO.unit
+            //TODO Storages
+            case Created(_, _, _, _, types, _, _, _) if types.contains(StorageType)        =>
+              IO.unit
+            case Updated(_, _, _, _, types, _, _, _) if types.contains(StorageType)        =>
+              IO.unit
+            case Deprecated(_, _, _, _, types, _, _) if types.contains(StorageType)        =>
+              IO.unit
+            // Data resources
+            case Created(id, _, _, schema, _, source, _, _)                                =>
+              val schemaSegment           =
+                if (schema.original == unsconstrained) IriSegment(Vocabulary.schemas.resources)
+                else IriSegment(schema.original)
+              val fixedSource             = fixSource(source)
+              def createResource(s: Json) = resources.create(IriSegment(id), projectRef, schemaSegment, s)
+              createResource(fixedSource)
+                .onErrorRecover { case _: ResourceRejection.InvalidJsonLdFormat =>
+                  createResource(fixId(fixedSource, id))
+                }
+                .toTask
+                .void
+            case Updated(id, _, _, _, _, source, _, _)                                     =>
+              val fixedSource             = fixSource(source)
+              def updateResource(s: Json) = resources.update(IriSegment(id), projectRef, None, cRev, s)
+              updateResource(fixedSource)
+                .onErrorRecover { case _: ResourceRejection.InvalidJsonLdFormat =>
+                  updateResource(fixId(fixedSource, id))
+                }
+                .toTask
+                .void
+            case Deprecated(id, _, _, _, _, _, _)                                          =>
+              resources.deprecate(IriSegment(id), projectRef, None, cRev).toTask.void
+            // Tagging
+            case TagAdded(id, _, _, _, targetRev, tag, _, _)                               =>
+              // No information on resource type in tag event :'(
+              resources
```

Comment:
I guess for now it is alright, but to do it for all resource types we will have to do it in another way, folding over a list or something like that.

Reply:
Yep, I have taken the easy path for now as we haven't decided how we will deal with storages/files/views.
I guess traits implemented by the plugins ?

Reply:
Yeah I guess so. So far we didn't need it for anything else, since the logic is handled by the routes. But we might need it for migrations